### DISCUSSION TITLE
Can't update from 2.2.19 to 2.2.20

### DISCUSSION THREAD

**Discussion by Swinto2000** [UtilitechAS/amsreader-firmware#594] [Q&A]
I have a ESP32 and try to update to latest fw, pressed the arrow "new version: v2.2.20" and also tried to upload file but after a while it comes back to the old version. Get error "Previous upgrade attempt (v2.2.20) does not match current version (v2.2.19) [255/255]". This has worked fine before, any ideas?
BR
Nils

**Comment by gskjold** [comment]
On error code 255/255, it has initiated the upgrade, but for some reason it has crashed before it could start the actual update. Not sure what that could be, the code used is pretty vanilla ESP. Did you get the same error code for both one-click upgrade and upload? Regarding uploading, you can check the serial console for output

**Reply by Swinto2000** [reply]
Got the error message when uploading. I'll try to check the serial console but I'm going away for a few days, maybe next week.
/N

**Reply by Swinto2000** [reply]
Turned off and then on my ESP32, rebooting didn't help and then update worked so the problem is solved, thanks for excellent program and help.

Best Regards
Nils
